src/coreclr/src/vm/weakreferencenative.cpp

```diff
@@ -102,17 +103,17 @@ struct WeakHandleSpinLockHolder
 
 #ifdef FEATURE_COMINTEROP
 
-// Get a WinRT weak reference for the object underlying an RCW if applicable.  If the incoming object cannot
-// use a WinRT weak reference, nullptr is returned.  Otherwise, an AddRef-ed IWeakReference* for the COM
+// Get a native COM weak reference for the object underlying an RCW if applicable.  If the incoming object cannot
+// use a native COM weak reference, nullptr is returned.  Otherwise, an AddRef-ed IWeakReference* for the COM
 // object underlying the RCW is returned.
 //
-// In order to qualify to be used with a HNDTYPE_WEAK_WINRT, the incoming object must:
+// In order to qualify to be used with a HNDTYPE_WEAK_NATIVE_COM, the incoming object must:
 //  * be an RCW
 //  * respond to a QI for IWeakReferenceSource
 //  * succeed when asked for an IWeakReference*
 //
 // Note that *pObject should be GC protected on the way into this method
-IWeakReference* GetWinRTWeakReference(OBJECTREF* pObject)
+IWeakReference* GetComWeakReference(OBJECTREF* pObject)
```

Comment:
Note that this is non-pay for play API when called from WeakReference constructor. What is this change going to do to performance of WeakReference constructor when called on regular COM object?

Reply:
When it's called by a regular COM object, it will go down the first if path and have basically the same performance characteristics as it did previously. Only non-builtin-COM objects will go down the else path that does the additional work.